cli/command_benchmark_compression.go

```diff
@@ -106,13 +128,49 @@ func (c *commandBenchmarkCompression) run(ctx context.Context) error {
 		}
 	}
 
-	log(ctx).Infof("Repeating %v times per compression method (total %v). Override with --repeat=N.", repeatCount, units.BytesString(int64(repeatCount*len(data))))
+	algorithms := map[compression.Name]compression.Compressor{}
 
 	for name, comp := range compression.ByName {
-		if compression.IsDeprecated[name] && !c.deprecated {
-			continue
+		if c.shouldIncludeAlgorithm(name) {
+			algorithms[name] = comp
+		}
+	}
+
+	log(ctx).Infof("Will repeat each benchmark %v times per compression method (total %v). Override with --repeat=N.", repeatCount, units.BytesString(int64(repeatCount*len(data))))
+
+	switch c.operations {
+	case "compress":
+		benchmarkCompression = true
+		benchmarkDecompression = false
+	case "decompress":
+		benchmarkCompression = false
+		benchmarkDecompression = true
+	default:
+		benchmarkCompression = true
+		benchmarkDecompression = true
+	}
+
+	if benchmarkCompression {
+		if err := c.runCompression(ctx, data, repeatCount, algorithms); err != nil {
+			return err
 		}
+	}
+
+	if benchmarkDecompression {
+		if err := c.runDecompression(ctx, data, repeatCount, algorithms); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (c *commandBenchmarkCompression) runCompression(ctx context.Context, data []byte, repeatCount int, algorithms map[compression.Name]compression.Compressor) error {
+	var results []compressionBechmarkResult
+
+	log(ctx).Infof("Compressing input file %q (%v) using all compression methods.", c.dataFile, units.BytesString(int64(len(data))))
```

Comment:
is it still the case that _all_ compression methods are used here?

Reply:
replaced "all" with a number.